core/src/main/java/org/mule/runtime/core/internal/exception/GlobalErrorHandler.java

```diff
@@ -24,7 +24,7 @@
   }
 
   public ErrorHandler createLocalErrorHandler(Location flowLocation) {
-    ErrorHandler local = new ErrorHandler();
+    ErrorHandler local = new LocalErrorHandler();
```

Comment:
Notice that you also have to use the system property here because with the former behavior the exceptionListeners should be stopped.